modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala

```diff
@@ -916,6 +1057,42 @@ object ObservationService {
         FROM t_observation
         WHERE c_observation_id IN (
       """.apply(gid, index) |+| which |+| void")"
+
+    def getProgramCfpId(pid: Program.Id): AppliedFragment =
+      sql"""
+        SELECT c_cfp_id
+        FROM t_proposal
+        WHERE c_program_id = $program_id
+      """.apply(pid)
```

Comment:
A minor style thing but what is your opinion on making this, say:

```scala
val SelectCfp: Query[Program.Id, CallForProposals.Id] =
  sql"""
    SELECT c_cfp_id
    FROM t_proposal
    WHERE c_program_id = $program_id
  """.query(cfp_id)
```

Then you can call it with a one-liner like `session.option(Statements.SelectCfp)(pid)`.  I've been trying to do it that way when possible since `AppliedFragment` doesn't really signal what the parameters and output are expected to be.

Reply:
That does seem clearer. I'll switch.

Reply:
I guess some of these could be moved to the `CallForProposalsService` to keep them together with other CfP stuff.

Reply:
I thought about that, but the queries seem pretty specific to the task at hand. If we move them, we'd probably want to make them more general, which would require a CallForProposals class, which we currently don't have. That said, I'm not opposed to doing that.